experimental/packages/opentelemetry-instrumentation-fetch/test/fetch.test.ts

```diff
@@ -531,20 +524,24 @@ describe('fetch', () => {
       ]);
     });
 
-    it('should set trace headers', () => {
+    it('should set trace headers', async () => {
       const span: api.Span = exportSpy.args[1][0][0];
+      assert.ok(lastResponse instanceof Response);
+
+      const { request } = await lastResponse.json();
+
       assert.strictEqual(
-        lastResponse.headers[X_B3_TRACE_ID],
+        request.headers[X_B3_TRACE_ID],
```

Comment:
The previously badly named variable made this very confusing, but this is actually asserting that we made the _request_ with the correct headers. Specifically that when the URL is not ignored, the instrumentation code will add these additional headers (which then gets put into the mock server response).